### PR TITLE
optional dependencies on pip

### PR DESCRIPTION
**Pull request by hmaarrfk**
Just a warning, there is no wheel file for [`imread`](https://pypi.org/project/imread/#files), an optional dependency, so there is virtually 0 way that this will work on anybody's machine....

Test on your machine
```
conda create --name sk-test-pip-optional --offline --yes python=3
conda activate sk-test-pip-optional
pip install numpy cython
pip install -e ".[test]"  --disable-pip-version-check --no-build-isolation
```

Output:
```
[...] # long stuff
Successfully installed codecov-2.0.15 coverage-4.5.2 flake8-3.6.0 mccabe-0.6.1 pycodestyle-2.4.0 pyflakes-2.0.0 pytest-cov-2.6.1 scikit-image
```

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](https://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by jni** [comment]
@baerbock please refrain from making such comments on any scikit-image forum. Please refer to our [community guidelines](https://scikit-image.org/community_guidelines.html) and only make constructive criticism.

Additionally, regardless of any *valid* criticism you might have of conda and pip (the above clearly isn't), they are the de facto standard distribution channels in Python and we *must* support them.

File: setup.py [hunk 1]
```diff
@@ -112,8 +112,20 @@ def build_extensions(self):
...
+        requires = [l.strip() for l in fid.readlines() if l]
+
+    return requires
+
+
+INSTALL_REQUIRES = parse_requirements_file('requirements/default.txt')
+extras_require = {
+    'docs': parse_requirements_file('requirements/docs.txt'),
+    'optional': parse_requirements_file('requirements/optional.txt'),
+    'test': parse_requirements_file('requirements/test.txt'),
```
Comment (hmaarrfk) [inline setup.py]: i'm not too sure about this test dependency.

Comment (soupault) [reply setup.py]: Same can be said for `docs` :). I'm fine with the current version.

**Comment by hmaarrfk** [comment]
@scikit-image/core can we drop the dependency on `imread`??? It is considered beta software without a wheel on pypi.

**Comment by jni** [comment]
I don't have any insight about the imread dependency. @stefanv? I'd be in favour of dropping it since Pillow seems to cover most of our bases.

**Comment by soupault** [comment]
@jni why Pillow? I though we're going to switch eventually to imageio - https://github.com/scikit-image/scikit-image/issues/1605.

**Comment by hmaarrfk** [comment]
"not depending" on pillow is a little moot since "imageio" depends on it

the default imageio backend is pillow. They just figured out how to avoid a memory copy for pillow.

I went down the rabbit hole a while back and figured out that imread was using pillow quite effectively:
https://github.com/hmaarrfk/redpil
(see benchmarks at bottom).

**Comment by jni** [comment]
@soupault what @hmaarrfk said. And anyway the point I was making is that we *do* depend on pillow right now (see `requirements/default.txt`), so probably we don't need to depend on `imread` *right now*. @stefanv what do you think about removing this dependency?

@hmaarrfk very cool benchmarks, and I appreciate the multi-level pun there! =D

File: setup.py [hunk 2]
```diff
@@ -112,8 +112,19 @@ def build_extensions(self):
...
+    with open(filename) as fid:
+        requires = [l.strip() for l in fid.readlines() if l]
+
+    return requires
+
+
+INSTALL_REQUIRES = parse_requirements_file('requirements/default.txt')
+extras_require = {
+    dep : parse_requirements_file('requirements/' + dep + '.txt')
+    for dep in ['docs', 'optional', 'test']
```
Comment (sciunto) [inline setup.py]: Perhaps, a comment to say that extra is omitted on purpose.

Comment (hmaarrfk) [reply setup.py]: I hope that one helps! thanks for the input.

**Pull request merged**